### DISCUSSION TITLE
navigation tree, based on file name containing dot and optionally virtual file names inside files or in sidecar xmp files

### DISCUSSION THREAD

**Discussion by aisbergde** [gsantner/markor#2228] [Ideas] [answered]
To classify my notes I use filenames like

- topic1.subtopic2.subtopic3.md
- topic1.subtopic4.adoc
- topic5.txt

I started to use this syntax a few years ago, when Markor only supported search in file names but not in file content.

Having a lot of notes makes the navigation anyway a bit hard.

My idea is an additional nested navigation tree, based on this dot based naming. In the example the tree would be

```
topic1
  subtopic2
    subtopic3
  subtopic4
topic5
```

additionally I would suggest to use a syntax inside the file content to allow to add additional virtual file names, following the same convention. And these entries should be used to place the file additionally in other places in the tree. Therefore a file can have one real file name and optionally multiple virtual filenames.

For example, if the first file gets an additional virtual filename
`topic1.subtopic2.subtopic6` then it will be shown in 2 places in the tree.

The implementation of the virtual filename syntax could or should be different per content type (Markdown, AsciiDoc, ...). For example in Markdown this could be a specific comment or something inside a yaml header, AsciiDoc has document properties.

I know that there are valid arguments against scanning file content for additional metadata. But I have not so much ideas, how to add additional metadata. An idea would be something similar to exif or xmp data in photos, which can be embedded into the file or can be provided in sidecar files. sidecar files could be an interesting option, because it would avoid the need to scan the main files content. The common and known sidecar XMP file format could be used. There could be just additional files like
`topic1.subtopic2.subtopic3.md.xmp`

**Comment by gsantner** [comment, marked as answer]
No plan to make an additional virtual abstraction to filename/folders.

However you can search for files and their contents, with/without regex, as well have it in subfolders or not.

I personally would keep the categorization to certain degee, but move off files into subfolders still in meaningful way. (i.e. don't have a folder with thousand files). Allows to find it in all other applications & filemanagers easy too.

**Reply by aisbergde** [reply]
In General search works fine, in filenames and in content. But it's not supported to get a dynamic result list after some input of some letters of the search string. Are there any ideas how to use an index based on file names? To enter at least a few letters and then to bring a dynamic result set?

Sometimes I am using an external file explorer for dynamic search and then I open the file in Markor.

The final goal of course would be a kind of tagging system. All these methods are workarounds to define and find tagged content.

**Reply by jawadakhan97** [reply]
on macos apple implenents tags which i use extensively for papers.

not sure how they implement it, is there a way to have an external file, say .index or .bib which is a plaintext format with the kind of metadata for the files in the same directory?

i'm sure someone has already done this, to give a quick idea with a file directory like this:

＋root
｜-- .index
｜--  01--2024-01_todo.txt
｜--  42--draft-eng-paper.md  
｜--  30--list-fav-keyboards.md
｜--  21--chase-bank-statement.csv
｜--  34--warhammer-strats.md
｜--  31--whale-song-rhymes.txt
｜--  42--final-presentation-v3-last.pptx
｜--  42--final-presentation-v3-last-script.md

```
METADATA
00 - PERSONAL
     01 - TODO
10 - WORK
     11 - DAY JOB
     12 - SIDE JOB
     13 - DROPSHIPPING BUSINESS
20 - FINANCE
     21 - TAX
     22 - INVESTMENT
     23 - SAVINGS
     24 - DEBT
30 - HOBBY
     31 - MUSIC PRODUCTION
     32 - PAINTING
     33 - D&D
     34 - WARHAMMER 40K
40 - SCHOOL
     41 - UNDERGRAD
     42 - MASTERS
```

The content can be annotated in-app and it wouldn't really be 'kept back' by markor which really strives to have compatibility and portability (which makes it such a great app, ty devs 💖)

By annotated I mean the tags are visible in-app as a prepend to the file name, or markor could adjust how the filenames are presented, idk by color coding each category.

serarching by tag with a flag or just by using the codes could work

let me know what you think! i'm trying to figure something out myself and 2 heads are better than 1

**Reply by aisbergde** [reply]
Are there any tools to maintain this kind of files containing meta data?

In General it could be a good idea to use optional metadata files, if they exists. Then Markor would not be responsible for creating and maintaining them, but should use only existing content. This format, or other formats, any way to define virtual trees.
